servers/audio_server.cpp

```diff
@@ -1618,6 +1618,20 @@ void AudioServer::update() {
 	}
 	bus_details_graveyard.maybe_cleanup();
 	bus_details_graveyard_frame_old.maybe_cleanup();
+
+	// TODO: Reverse the process, instead of polling if the sample playback is active,
+	//       make JavaScript callback Godot when a sample playback is removed.
+	if (!sample_playback_list.is_empty()) {
+		LocalVector<Ref<AudioSamplePlayback>> sample_playbacks_to_remove;
+		for (Ref<AudioSamplePlayback> &sample_playback : sample_playback_list) {
+			if (!is_sample_playback_active(sample_playback)) {
+				sample_playbacks_to_remove.push_back(sample_playback);
+			}
+		}
+		for (Ref<AudioSamplePlayback> &sample_playback_to_remove : sample_playbacks_to_remove) {
+			sample_playback_list.erase(sample_playback_to_remove);
+		}
```

Comment:
Is there a better way to remove elements from the currently looping for loop list?

Reply:
Just loop with an index instead of range-based `for`, and start with the last element.

Reply:
Finally, the callback functionality existed, but wasn't working. So I fixed it, and removed that part.
But I'll remember the trick. Thanks @rburing!